tmail-backend/rate-limiter/src/main/scala/com/linagora/tmail/rate/limiter/api/RateLimitation.scala

```diff
@@ -0,0 +1,83 @@
+package com.linagora.tmail.rate.limiter.api
+
+import com.linagora.tmail.rate.limiter.api.RateLimitation.AllowedQuantity
+import eu.timepit.refined.api.Refined
+import eu.timepit.refined.numeric.Positive
+
+import java.time.Duration
+import java.util.UUID
+import scala.util.Try
+
+object RateLimitation {
+  type AllowedQuantity = Long Refined Positive
+}
+
+sealed trait LimitType {
+  def asString(): String
+
+  def allowedQuantity(): AllowedQuantity
+}
+
+case class Count(quantity: AllowedQuantity) extends LimitType {
+  override def asString(): String = "count"
+
+  override def allowedQuantity(): AllowedQuantity = quantity
+}
+
+case class Size(quantity: AllowedQuantity) extends LimitType {
+  override def asString(): String = "size"
+
+  override def allowedQuantity(): AllowedQuantity = quantity
+}
+
+case class RateLimitation(name: String, period: Duration, limit: Seq[LimitType])
+
+object RateLimitingPlanId {
+  def generate: RateLimitingPlanId = RateLimitingPlanId(UUID.randomUUID())
+
+  def from(value: String): Try[RateLimitingPlanId] = Try(RateLimitingPlanId(UUID.fromString(value)))
+}
+
+case class RateLimitingPlanId(value: UUID)
+
+case class RateLimitingPlanName(value: String)
+
+sealed trait ProcessorLimit {
```

Comment:
I would like use to find a better name than ProcessorLimit

Reply:
`TransitionType` is better?